src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -59,42 +67,104 @@ export class HuiDialogEditCard extends LitElement {
         paper-dialog {
           width: 650px;
         }
+        .element-editor {
+          margin-bottom: 16px;
+        }
       </style>
       <paper-dialog with-backdrop>
         <h2>Card Configuration</h2>
         <paper-dialog-scrollable>
-          <hui-yaml-editor
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            // Look to see if this has already be done if yes dont use until just render this._elementConfig
+            this._showUIEditor && !this._elementConfig
+              ? html`<div class="element-editor">${until(
+                  this._loadConfig().then(() => this._loadElementConfig()),
+                  html`<span>Loading...</span>`
+                )}</div>`
+              : this._showUIEditor
+                ? html`<div class="element-editor">${this._elementConfig}</div>`
+                : html`
+                  <hui-yaml-editor
+                    .yaml="${this._cardConfig}"
+                    @yaml-changed="${this._handleYamlChanged}"
+                  ></hui-yaml-editor>`
+          }
           <hui-yaml-card-preview
             .hass="${this.hass}"
             .yaml="${this._cardConfig}"
           ></hui-yaml-card-preview>
         </paper-dialog-scrollable>
         <div class="paper-dialog-buttons">
-          <paper-button @click="${this._closeDialog}">Cancel</paper-button>
-          <paper-button @click="${this._updateConfig}">Save</paper-button>
+          <paper-button
+            @click="${this._toggleEditor}"
+          >Toggle Editor</paper-button>
+          <paper-button
+            @click="${this._closeDialog}"
+          >Cancel</paper-button>
+          <paper-button
+            @click="${this._updateConfig}"'
+          >Save</paper-button>
         </div>
       </paper-dialog>
     `;
   }
 
   private _handleYamlChanged(ev) {
-    this._previewEl.yaml = ev.detail.yaml;
+    this._handleConfigChanged("yaml", ev.detail.yaml);
+  }
+
+  private _handleConfigChanged(type: string, changedProps: any): void {
```

Comment:
This method should also store the config in case we want to save it ?

Reply:
Also, `changedProps` ? shouldn't it just be `value` or `config`.

Also, you can do better than `any`. Specify all types we expect.

Reply:
hmm that would rewrite the config each time a letter or anything is changed. Is that better than just storing the config on save click?

Reply:
Probably. I just realized that `Config | String` would work here. Thats what that syntax means. IT can be either of those types.. AS you can see Im new to types

Reply:
if you don't create a Lit property, you can store it as `this._configDraft` or whatever and it won't trigger a re-render. Lit ain't magic… just rerenders on props we tell to re-render on

Reply:
I see that saving it each time will make saving easier